dotcom-rendering/src/web/components/elements/YoutubeBlockComponent.tsx

```diff
@@ -82,13 +86,39 @@ export const YoutubeBlockComponent = ({
 	expired,
 	role,
 	adTargeting,
-	consentState,
 	isMainMedia,
 	height = 259,
 	width = 460,
 	duration,
 	origin,
 }: Props): JSX.Element => {
+	const [consentState, setConsentState] = useState<ConsentState | undefined>(
+		undefined,
+	);
+
+	useOnce(() => {
```

Comment:
This is just a hunch... although this will initiate CMP and set consent state on a second pass it won't block the render of `YoutubeAtom` on the first pass without consent.

`YoutubeAtom` will render its first and second pass (after its internal useEffect) without consent and therefore default to disabled ads.

Previously the `HydrateOnce` for `YoutubeBlockComponent` would `waitFor={[consentState]}`

I _think_...

Reply:
Is is simply that consentState should be added to the dependency array on L120?

Reply:
Thanks @arelra. I think you're right. Previously we waited for `consentState` in `App.tsx` using a `waitFor` but we're not doing that here. I think I saw it suggested that we could fix this by making `consentState` a dependency for `useOnce` to run, which I think it true. This would essentially replicate the logic we had previously. Ahead of that though, I'm investigating if there are some optimisations that can be made in the `YouTubeAtom` itself. I'll update back here when I've done this.

Reply:
You might be already considering this but I think _completely_ separating the overlay from the player will help reduce complexity of `YoutubeAtom` quite a bit and allow blocking of only whats required - i.e. instantiation of the player.

It will just need a method to trigger the play from the overlay to the player.

Reply:
Based on this discussion, I looked deeper at how we render You Tube videos and discovered that we can improve how we do this.

@mchv pointed out [a great lib](https://github.com/justinribeiro/lite-youtube) to me earlier which defers the loading of the you tube javascript until the user clicks the poster overlay. My first response was to say that we already do this but when I looked more I realised we don't! So I've added this feature to our `YouTubeAtom`

https://github.com/guardian/atoms-rendering/pull/303

Whilst doing this I also added a defer to wait for `consentState` which, if we're happy with that approach` would address the issues in this thread, without the need to manage waiting for state locally.

I'm also happy to put the `waitFor` in and not have `YouTubeAtom` manage this (we could make it required) or some combination of the both. Whichever api makes the best sense to the group is fine by me.

Reply:
Is `useOnce` required if the dependency array is `[]` ?
If not you could just have a useEffect with `[]` ?